File: web/src/api/adEvents.js
import service from '@/utils/request'

// @Tags AdEvents
// @Summary 创建AdEvents
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.AdEvents true "创建AdEvents"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /adEvents/createAdEvents [post]
export const createAdEvents = (data) => {
     return service({
         url: "/adEvents/createAdEvents",
         method: 'post',
         data
     })
 }


// @Tags AdEvents
// @Summary 删除AdEvents
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.AdEvents true "删除AdEvents"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /adEvents/deleteAdEvents [delete]
 export const deleteAdEvents = (data) => {
     return service({
         url: "/adEvents/deleteAdEvents",
         method: 'delete',
         data
     })
 }

// @Tags AdEvents
// @Summary 删除AdEvents
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.IdsReq true "批量删除AdEvents"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /adEvents/deleteAdEvents [delete]
 export const deleteAdEventsByIds = (data) => {
     return service({
         url: "/adEvents/deleteAdEventsByIds",
         method: 'delete',
         data
     })
 }

// @Tags AdEvents
// @Summary 更新AdEvents
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.AdEvents true "更新AdEvents"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"更新成功"}"
// @Router /adEvents/updateAdEvents [put]
 export const updateAdEvents = (data) => {
     return service({
         url: "/adEvents/updateAdEvents",
         method: 'put',
         data
     })
 }


// @Tags AdEvents
// @Summary 用id查询AdEvents
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.AdEvents true "用id查询AdEvents"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"查询成功"}"
// @Router /adEvents/findAdEvents [get]
 export const findAdEvents = (params) => {
     return service({
         url: "/adEvents/findAdEvents",
         method: 'get',
         params
     })
 }


// @Tags AdEvents
// @Summary 分页获取AdEvents列表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.PageInfo true "分页获取AdEvents列表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /adEvents/getAdEventsList [get]
 export const getAdEventsList = (params) => {
     return service({
         url: "/adEvents/getAdEventsList",
         method: 'get',
         params
     })
 }

File: server/service/adEvents.go
package service

import (
	"fmt"
	"gin-vue-admin/global"
	"gin-vue-admin/model"
	"gin-vue-admin/model/request"
	"gin-vue-admin/utils"
)

// @title    CreateAdEvents
// @description   create a AdEvents
// @param     adEvents               model.AdEvents
// @auth                     （2020/04/05  20:22）
// @return    err             error

func CreateAdEvents(adEvents model.AdEvents) (err error) {
	err = global.GVA_DB_LOGIC.Create(&adEvents).Error
	return err
}

// @title    DeleteAdEvents
// @description   delete a AdEvents
// @auth                     （2020/04/05  20:22）
// @param     adEvents               model.AdEvents
// @return                    error

func DeleteAdEvents(adEvents model.AdEvents) (err error) {
	err = global.GVA_DB_LOGIC.Delete(adEvents).Error
	return err
}

// @title    DeleteAdEventsByIds
// @description   delete AdEventss
// @auth                     （2020/04/05  20:22）
// @param     adEvents               model.AdEvents
// @return                    error

func DeleteAdEventsByIds(ids request.IdsReq) (err error) {
	err = global.GVA_DB_LOGIC.Delete(&[]model.AdEvents{}, "id in ?", ids.Ids).Error
	return err
}

// @title    UpdateAdEvents
// @description   update a AdEvents
// @param     adEvents          *model.AdEvents
// @auth                     （2020/04/05  20:22）
// @return                    error

func UpdateAdEvents(adEvents *model.AdEvents) (err error) {
	err = global.GVA_DB_LOGIC.Save(adEvents).Error
	return err
}

// @title    GetAdEvents
// @description   get the info of a AdEvents
// @auth                     （2020/04/05  20:22）
// @param     id              uint
// @return                    error
// @return    AdEvents        AdEvents

func GetAdEvents(id uint) (err error, adEvents model.AdEvents) {
	err = global.GVA_DB_LOGIC.Where("id = ?", id).First(&adEvents).Error
	return
}

// @title    GetAdEventsInfoList
// @description   get AdEvents list by pagination, 分页获取AdEvents
// @auth                     （2020/04/05  20:22）
// @param     info            PageInfo
// @return                    error

func GetAdEventsInfoList(info request.AdEventsSearch) (err error, list interface{}, total int64) {
	limit := info.PageSize
	offset := info.PageSize * (info.Page - 1)
	// 创建db
	db := global.GVA_DB_LOGIC.Model(&model.AdEvents{})
	var adEventss []model.AdEvents
	// 如果有条件搜索 下方会自动创建搜索语句
	if info.Platform != "" {
		db = db.Where("`platform` = ?", info.Platform)
	}
	if info.AdId != "" {
		db = db.Where("`ad_id` LIKE ?", "%"+info.AdId+"%")
	}
	if info.AdType != "" {
		db = db.Where("`ad_type` = ?", info.AdType)
	}
	if info.AdAction != "" {
		db = db.Where("`ad_action` = ?", info.AdAction)
	}
	if info.AdChannel != "" {
		db = db.Where("`ad_channel` = ?", info.AdChannel)
	}
	if !info.CreateTime.IsZero() {
		db = db.Where("`create_time` <> ?", info.CreateTime)
	}
	if !info.UpdateTime.IsZero() {
		db = db.Where("`update_time` <> ?", info.UpdateTime)
	}
	if len(info.Order) > 0 && len(info.OrderBy) > 0 {
		db = db.Order(fmt.Sprintf("%s %s", utils.Camel2Case(info.Order), info.OrderBy))
	} else {
		db = db.Order("id desc")
	}
	err = db.Count(&total).Error
	err = db.Limit(limit).Offset(offset).Find(&adEventss).Error
	return err, adEventss, total
}
